narwhal/executor/src/notifier.rs

```diff
@@ -45,31 +44,29 @@ impl<State: ExecutionState + Send + Sync + 'static> Notifier<State> {
                 batch.transactions.len()
             );
             self.metrics.notifier_processed_batches.inc();
+
             let mut bytes = 0usize;
             for (transaction_index, transaction) in batch.transactions.into_iter().enumerate() {
                 let execution_indices = ExecutionIndices {
                     next_certificate_index: index.next_certificate_index,
                     next_batch_index: index.batch_index + 1,
                     next_transaction_index: transaction_index as u64 + 1,
+                    last_committed_leader: index.sub_dag.leader.round(),
                 };
                 bytes += transaction.len();
+
+                // NOTE: We now have access to the entire commit through `index.sub_dag`. We may
+                // thus choose to make it available to Sui through `handle_consensus_transaction`.
+
                 self.callback
-                    .handle_consensus_transaction(
-                        &index.consensus_output,
-                        execution_indices,
-                        transaction,
-                    )
+                    .handle_consensus_transaction(&index.output, execution_indices, transaction)
```

Comment:
Here I would change the interface to pass in the whole commit. And allow Sui to process the whole commit as a whole. There is no point feeding Sui or anyone else a commit at a time, and having to manage this process if it gets interrupted half way through.

Reply:
totally agree, I planned to do it in a separate PR to keep things short